odoo/models.py

```diff
@@ -1637,7 +1637,17 @@ def _name_search(self, name='', args=None, operator='ilike', limit=100, name_get
         # optimize out the default criterion of ``like ''`` that matches everything
         elif not (name == '' and operator in ('like', 'ilike')):
             aggregator = expression.AND if operator in expression.NEGATIVE_TERM_OPERATORS else expression.OR
-            domain = aggregator([[(field_name, operator, name)] for field_name in search_fnames])
+            domains = []
+            for field_name in search_fnames:
+                root_field_name = field_name.split('.')[0]
+                if self._fields[root_field_name].relational or operator == "in":
+                    domains.append([(field_name, operator, name)])
+                else:
+                    try:
+                        domains.append([(field_name, operator, self._fields[field_name].convert_to_cache(name, self, validate=False))])
```

Comment:
This doesn't make sense to me, as the values in domains are not in the same format as the cache format.

Reply:
I think that you should use `convert_to_write()` instead.
That would make more sense.

Reply:
Thanks for the insight, it seems to have solved the issue we had with list of values being passed to the search, so it's way better that way :)